browser/todo-list.md

```diff
@@ -3,291 +3,248 @@
+The data for all todos are saved as JSON to the Gaia hub linked to your Secret Key using the [`putFile`](http://blockstack.github.io/blockstack.js/globals.html#putfile) method of the `userSession` object in the [`src/assets/data-store.js`](https://github.com/blockstack/blockstack-todos/blob/master/src/assets/data-store.js#L26) module:
 
 ```js
-import React from 'react';
-import { useConnect } from '@blockstack/connect';
-
-export const Signin = () => {
-  const { doOpenAuth } = useConnect();
-
-  return (
-    <button
-      onClick={() => doOpenAuth()}
-    >
-      Sign In with Blockstack
-    </button>
-  )
+export const saveTasks = async (userSession, tasks, isPublic) => {
+  await userSession.putFile(TASKS_FILENAME, JSON.stringify({ tasks, isPublic }), {
+    encrypt: !isPublic,
+  });
 };
 ```
 
-Once the user authenticates, the application handles the `authResponse` in the `src/components/Profile.js` file. :
+These todos are subsequently loaded using the [`getFile`](http://blockstack.github.io/blockstack.js/globals.html#getfile) method of the same object in the same module:
 
 ```js
-...
-componentWillMount() {
-    if (userSession.isSignInPending()) {
-      userSession.handlePendingSignIn().then((userData) => {
-        //if (!userData.username) {
-        //  throw new Error('This app requires a username.')
-        //}
-        window.location = window.location.origin;
-      });
-    }
-  }
-...
+export const fetchTasks = async (userSession, username) => {
+  const tasksJSON = await userSession.getFile(TASKS_FILENAME, {
+    decrypt: false,
+    username: username || undefined,
+  });
+  // code to format and return the tasks
+};
 ```
 
-If [`isUserSignedIn()`](https://blockstack.github.io/blockstack.js/#isusersignedin) is true, the user was previously signed in so Blockstack pulls the data from the browser and uses it in our application. If the check on  [`UserSession.isSignInPending()`](https://blockstack.github.io/blockstack.js/#issigninpending) is true, a previous `authResponse` was sent to the application but hasn't been processed yet. The `handlePendingSignIn()` function processes any pending sign in.
+By default, the `putFile` and `getFile` methods automatically encrypt data when saved and decrypt it when retrieved, using the user's Secret Key. This ensures that only the user has the ability to view this data.
 
-Signout is handled in `src/components/App.js`.
+When deleting a todo, the same `putFile` method is used to save a new JSON array of todos that excludes the deleted todo.
 
-```js
-  handleSignOut(e) {
-    e.preventDefault();
-    userSession.signUserOut(window.location.origin);
-  }
-```
+### Publish your todos publicly
 
-The method allows the application creator to decide where to redirect the user upon Sign Out:
+Select "Make public" to make your todos accessible to the public for sharing via URL.
```

Comment:
I'd point out that one should add a few todos.. I skipped that and opened up a 404 later